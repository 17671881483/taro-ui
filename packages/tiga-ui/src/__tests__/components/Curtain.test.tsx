import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent, render } from '@testing-library/react'
import { Image } from '../__mock__/taroComponent'

describe('AtCurtain 测试在 weapp 环境，h5类似 -> ', () => {
  const OLD_ENV = { ...process.env }
  process.env.TARO_ENV = 'weapp'
  const curtainPng = 'https://taro-ui.aotu.io/h5/static/images/curtain.png'

  afterAll(() => {
    process.env = { ...OLD_ENV } // Restore old environment
    jest.resetModules()
  })

  afterEach(cleanup)

  const AtCurtain = jest.requireActual('../../components/curtain').default

  it('展示：默认基础幕帘', () => {
    const component = renderer.create(
      <AtCurtain isOpened={true}>
        <Image style={{ width: '100%' }} mode='widthFix' src={curtainPng} />
      </AtCurtain>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('展示：幕帘关闭按钮展示在指定位置', () => {
    const component = renderer.create(
      <AtCurtain isOpened={true} closeBtnPosition='top'>
        <Image style={{ width: '100%' }} mode='widthFix' src={curtainPng} />
      </AtCurtain>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('交互：点击关闭图标关闭幕帘', () => {
    const onCloseFn = jest.fn()

    const { getByTestId } = render(
      <AtCurtain isOpened={true} onClose={onCloseFn}>
        <Image style={{ width: '100%' }} mode='widthFix' src={curtainPng} />
      </AtCurtain>
    )

    const closeBtn = getByTestId('at-curtain__btn-close')
    const container = getByTestId('at-curtain')

    expect(container.className).not.toContain('closed')
    fireEvent.click(closeBtn)
    expect(container.className).toContain('closed')
    expect(onCloseFn).toBeCalledTimes(1)
  })
})
