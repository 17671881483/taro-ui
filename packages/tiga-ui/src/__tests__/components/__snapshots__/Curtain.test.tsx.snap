// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCurtain 测试在 weapp 环境，h5类似 ->  展示：幕帘关闭按钮展示在指定位置 1`] = `
<view
  className="at-curtain"
  data-testid="at-curtain"
  onClick={[Function]}
>
  <view
    className="at-curtain__container"
  >
    <view
      className="at-curtain__body"
    >
      <image
        mode="widthFix"
        src="https://taro-ui.aotu.io/h5/static/images/curtain.png"
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <view
        className="at-curtain__btn-close at-curtain__btn-close--top"
        data-testid="at-curtain__btn-close"
        onClick={[Function]}
      >
        <text
          className="at-icon at-icon-curtain_icon_cancel at-curtain__btn-close--at-icon"
          style={
            Object {
              "color": undefined,
              "fontSize": "36px",
            }
          }
        />
      </view>
    </view>
  </view>
</view>
`;

exports[`AtCurtain 测试在 weapp 环境，h5类似 ->  展示：默认基础幕帘 1`] = `
<view
  className="at-curtain"
  data-testid="at-curtain"
  onClick={[Function]}
>
  <view
    className="at-curtain__container"
  >
    <view
      className="at-curtain__body"
    >
      <image
        mode="widthFix"
        src="https://taro-ui.aotu.io/h5/static/images/curtain.png"
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <view
        className="at-curtain__btn-close at-curtain__btn-close--bottom"
        data-testid="at-curtain__btn-close"
        onClick={[Function]}
      >
        <text
          className="at-icon at-icon-curtain_icon_cancel at-curtain__btn-close--at-icon"
          style={
            Object {
              "color": undefined,
              "fontSize": "36px",
            }
          }
        />
      </view>
    </view>
  </view>
</view>
`;
